torch/testing/_internal/print_test_stats.py

```diff
@@ -24,6 +25,126 @@
 except ImportError:
     HAVE_BOTO3 = False
 
+
+# TODO: this is ugly duplicated code from run_test.py we should probably abstract out to common_utils.py
```

Comment:
Would common_utils be a good place for this, if we were to go with this solution?

Reply:
I'd say no, since we're planning on moving `print_test_stats.py` to `tools`; not sure how to solve the duplication though.